lucille-core/src/test/java/com/kmwllc/lucille/stage/ComputeFieldSizeTest.java

```diff
@@ -0,0 +1,79 @@
+package com.kmwllc.lucille.stage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import java.util.Arrays;
+import org.junit.Test;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.core.UpdateMode;
+
+public class ComputeFieldSizeTest {
+
+  private final StageFactory factory = StageFactory.of(ComputeFieldSize.class);
+
+  @Test
+  public void testNotAllRequiredFields() throws StageException {
+    assertThrows(StageException.class, () -> {
+      factory.get("ComputeFieldSizeTest/no-source.conf");
+    });
+    assertThrows(StageException.class, () -> {
+      factory.get("ComputeFieldSizeTest/no-destination.conf");
+    });
+  }
+
+  @Test
+  public void testSourceFieldDoesNotExist() throws StageException {
+    Stage stage = factory.get("ComputeFieldSizeTest/basic.conf");
+    Document doc1 = Document.create("doc1");
+
+    assertThrows(StageException.class, () -> {
+      stage.processDocument(doc1);
+    });
+  }
+
+  @Test
+  public void testSourceNotByteArray() throws StageException {
+    Stage stage = factory.get("ComputeFieldSizeTest/basic.conf");
+    Document doc1 = Document.create("doc1");
+    doc1.update("source", UpdateMode.OVERWRITE, "string");
+
+    assertThrows(NullPointerException.class, () -> {
```

Comment:
It's good that this test documents the current behavior -- but NPE's aren't great exceptions for stages to throw in general. To refine the behavior in this case, we have a choice between 1) skipping the doc, 2) adding something to the destination field to indicate the error (could be -1 to be consistent with a number going there, or could be a text error), 3) throwing a StageException. What do you think is best here? I'm leaning toward 3.